### PR TITLE
tree-sitter: 0.17.1 -> 0.17.3; run make install

### PR DESCRIPTION
**Pull request by colemickens**
###### Motivation for this change

Update `tree-sitter` to latest.

Also, this is necessary to actually install the lib/include/pkgconfig files. (neovim nightly now needs this and I know there are a number of us)

cc: @teto @mjlbach 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/parsing/tree-sitter/default.nix [hunk 1]
```diff
@@ -68,6 +68,11 @@ in rustPlatform.buildRustPackage {
     bash ./script/build-wasm --debug
   '';
 
+  postInstall = ''
+    export PREFIX=$out
+    make install
```
Comment (teto) [inline pkgs/development/tools/parsing/tree-sitter/default.nix:73]: that looks wrong. make install should be called by default. If you need to export PREFIX, it can be done via configureFlags.

Comment (teto) [reply pkgs/development/tools/parsing/tree-sitter/default.nix:73]: nevermind it uses buildRustPackage.

Comment (teto) [reply pkgs/development/tools/parsing/tree-sitter/default.nix:73]: this seems to install both static and shared libary. we should pick one. I think you (or another) could propose a patch for their Makefile to be more granular. Since homebrew is packaging it too, would be interesting to see how they've done it.

Comment (NickHu) [reply pkgs/development/tools/parsing/tree-sitter/default.nix:73]: In the arch community repo they package the shared library https://www.archlinux.org/packages/community/x86_64/tree-sitter/files/

**Comment by NickHu** [comment]
Overlay for the impatient: `~/.config/nixpkgs/overlays/tree-sitter.nix`

```nix
self: super:
{
  tree-sitter = super.tree-sitter.overrideAttrs (oldAttrs: {
    postInstall = "PREFIX=$out make install";
  });
}
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102763` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>tree-sitter</li>
  </ul>
</details>

File: pkgs/development/tools/parsing/tree-sitter/default.nix (hunk 1 above)
Comment (NickHu) [reply pkgs/development/tools/parsing/tree-sitter/default.nix]: I went to the liberty of addressing this in a follow-up commit; thanks for your contribution @colemickens!

@teto can you re-review and then I can merge?

**Comment by ipetkov** [comment]
I noticed that the metadata sets ```broken = stdenv.isDarwin || stdenv.isAarch64;```. I was able to build this successfully by adding ```buildInputs = lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];```, so it may be worth doing that in this PR.

File: pkgs/top-level/static.nix [hunk 2]
```diff
@@ -286,6 +286,8 @@ in {
 
   libexecinfo = super.libexecinfo.override { enableShared = false; };
 
+  tree-sitter = super.tree-sitter.override { static = true; };
```
Comment (alyssais) [inline pkgs/top-level/static.nix:289]: Is this right?

Comment (NickHu) [reply pkgs/top-level/static.nix]: Nice catch, fixed

File: pkgs/development/tools/parsing/tree-sitter/default.nix [hunk 3]
```diff
@@ -49,6 +51,8 @@ in rustPlatform.buildRustPackage {
   pname = "tree-sitter";
   inherit src version cargoSha256;
 
+  buildInputs = lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
```
Comment (NickHu) [inline pkgs/development/tools/parsing/tree-sitter/default.nix:54]: Not sure if this should be `darwin.apple_sdk.frameworks.Security` or `darwin.Security`; I have no way to test.

File: pkgs/development/tools/parsing/tree-sitter/default.nix [hunk 4]
```diff
@@ -1,8 +1,10 @@
-{ lib, stdenv
+{ lib, stdenv, darwin
 , fetchgit, fetchFromGitHub, fetchurl
 , writeShellScript, runCommand, which
 , rustPlatform, jq, nix-prefetch-git, xe, curl, emscripten
 , callPackage
+, enableShared ? true
+, enableStatic ? false
```
Comment (Profpatsch) [inline pkgs/development/tools/parsing/tree-sitter/default.nix:7]: `enableShared` what? Not really clear what that means.

Do we need both options? Why is it configurable?

File: pkgs/development/tools/parsing/tree-sitter/default.nix (hunk 3 above)
Comment (Profpatsch) [reply pkgs/development/tools/parsing/tree-sitter/default.nix:54]: What do other packages do?

Comment (Profpatsch) [reply pkgs/development/tools/parsing/tree-sitter/default.nix:54]: Or maybe: what prompted you to add it if you don’t have a MacOS machine?

File: pkgs/development/tools/parsing/tree-sitter/grammars/default.nix [hunk 5]
```diff
@@ -5,6 +5,7 @@
   embedded-template = (builtins.fromJSON (builtins.readFile ./tree-sitter-embedded-template.json));
   go = (builtins.fromJSON (builtins.readFile ./tree-sitter-go.json));
   html = (builtins.fromJSON (builtins.readFile ./tree-sitter-html.json));
+  java = (builtins.fromJSON (builtins.readFile ./tree-sitter-java.json));
```
Comment (Profpatsch) [inline pkgs/development/tools/parsing/tree-sitter/grammars/default.nix:8]: Can we put this into a separate commit?

Comment (Profpatsch) [reply pkgs/development/tools/parsing/tree-sitter/grammars/default.nix:8]: Same with the update of the grammars

**Review by Profpatsch — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/parsing/tree-sitter/default.nix (hunk 3 above)
Comment (jupblb) [reply pkgs/development/tools/parsing/tree-sitter/default.nix:54]: I can testify that the build process indeed fails without it on MacOS. This dependency seem to be typical for Rust packages, like `bat` or `git-and-tools.delta`.

Comment (ipetkov) [reply pkgs/development/tools/parsing/tree-sitter/default.nix:54]: @NickHu sorry I didn't clarify earlier, this is how I've normally seen this accomplished:
* add a parameter called `Security` at the top of `tree-sitter/default.nix`
* set `buildInputs = lib.optionals stdenv.isDarwin [ Security ];` in `tree-sitter/default.nix`
* update `top-level/all-packages.nix` with:

```
tree-sitter = callPackage ../development/tools/parsing/tree-sitter {
   inherit (darwin.apple_sdk.frameworks) Security;
};
```

You can ping me when the change is made and I can test that it builds on macOS :)

File: pkgs/development/tools/parsing/tree-sitter/default.nix (hunk 4 above)
Comment (NickHu) [reply pkgs/development/tools/parsing/tree-sitter/default.nix:7]: If you're building static, maybe you want to not have the shared libraries - configurable by this flag

File: pkgs/development/tools/parsing/tree-sitter/default.nix (hunk 3 above)
Comment (NickHu) [reply pkgs/development/tools/parsing/tree-sitter/default.nix]: @ipetkov ping

File: pkgs/development/tools/parsing/tree-sitter/default.nix (hunk 4 above)
Comment (NickHu) [reply pkgs/development/tools/parsing/tree-sitter/default.nix:7]: I had forgotten to set `enableShared = false` in `static.nix`; fixed now

File: pkgs/development/tools/parsing/tree-sitter/default.nix (hunk 3 above)
Comment (ipetkov) [reply pkgs/development/tools/parsing/tree-sitter/default.nix]: @NickHu 4d8adccab694baf4d413e8feef308c9b1da2d507 builds on macOS 👍

**Comment by NickHu** [comment]
Seems like this has gone quiet. I think I addressed everyone's concerns, so I'll just go ahead and merge this. If any follow-ups are required they can come in a separate PR
